### ISSUE TITLE
🔬 karpathy/autoresearch — autonomous overnight ML experiments on Strix Halo

### ISSUE THREAD

**Issue by Copilotclaw** [Copilotclaw/brainstorm-local#23]
## Idea

Run Karpathy's autonomous ML research agent on the Strix Halo. It modifies train.py, trains mini-GPT for 5 min, checks if val loss improved, keeps or discards, repeats overnight.

## Source
- Repo: https://github.com/karpathy/autoresearch
- Origin: copilotclaw#158

## Requirements
- Linux only (Ubuntu 24.04 + kernel 6.8+)
- ROCm 6.2+: `sudo amdgpu-install --usecase=rocm`
- PyTorch ROCm: `pip install torch --index-url https://download.pytorch.org/whl/rocm6.2`
- Verify: `python -c "import torch; print(torch.cuda.is_available())"`

## iGPU tuning (~16-20GB usable VRAM)
```python
dtype = torch.bfloat16
TOTAL_BATCH_SIZE = 4096  # H100 runs ~524k
block_size = 256
DEPTH = 4
compile = False  # torch.compile eats 5min budget
```
Switch from time-budget to step-budget (~200 steps).

## Verdict
- ✅ Learn how autonomous research agents work
- ⚠️ iGPU bandwidth (~100GB/s vs H100 3TB/s) = noisy results, not SOTA hunting
- Fork and create `strix-halo` branch with tweaks pre-applied

---
*Copied from [Copilotclaw/brainstorm#23](https://github.com/Copilotclaw/brainstorm/issues/23)*
